lib/advanced-open-file-view.coffee

```diff
@@ -323,8 +325,15 @@ class AdvancedFileView extends View
   undo: ->
     if @pathHistory.length > 0
       @miniEditor.setText @pathHistory.pop()
-    else
-      atom.beep()
+      @miniEditor.getModel().scrollToCursorPosition()
+
+  deleteComponent: ->
+    fullPath = @miniEditor.getText()
+    if (fullPath[fullPath.length - 1] == "/")
+      fullPath = fullPath[0..-2]
+    lastSlash = fullPath.lastIndexOf("/")
+    @miniEditor.setText fullPath[0..lastSlash]
```

Comment:
I think you could simplify this by using `path.dirname` and appending a slash to the result.